### PR TITLE
[Fix] `no-unknown-property` add properties onToggle, fill, as, pointer events, media events and touch events

### LINKED ISSUE #3384: [7.31.4] `no-unknown-property` throwing on valid props
Related to #3380

I'm seeing the following errors in my project:

```
Unknown property 'onToggle' found
```
`<details>` supports `onToggle`.

```
Invalid property 'fill' found on tag 'path', but it is only allowed on: svg
```
`<path>` supports `fill`.

```
Unknown property 'as' found
```
`<link>` supports `as`.

---

Edit: It looks like there are even more valid use cases that this rule is not covering. I'd suggest reverting the changes to the `no-unknown-property` rule for v7 and releasing it under a new major version. [React Known Props](https://github.com/tcodes0/react-known-props)  can be used to get all the known props, and users can use the `ignore` option on the rule to add other props like [`css`](https://github.com/jsx-eslint/eslint-plugin-react/issues/3384#issuecomment-1236100291).

### PR DESCRIPTION
**Pull request by sjarva**
Closes #3384

### CODE REVIEW

File: lib/rules/no-unknown-property.js [hunk 1]
```diff
@@ -230,7 +237,12 @@ function getDOMPropertyNames(context) {
   const ALL_DOM_PROPERTY_NAMES = DOM_PROPERTY_NAMES_TWO_WORDS.concat(DOM_PROPERTY_NAMES_ONE_WORD);
   // this was removed in React v16.1+, see https://github.com/facebook/react/pull/10823
   if (!testReactVersion(context, '>= 16.1.0')) {
-    return ['allowTransparency'].concat(ALL_DOM_PROPERTY_NAMES);
+    return ALL_DOM_PROPERTY_NAMES.concat(['allowTransparency']);
```
Comment (ljharb) [inline lib/rules/no-unknown-property.js:240]:
```suggestion
    return ALL_DOM_PROPERTY_NAMES.concat('allowTransparency');
```
what’s the reason to flip these?

File: lib/rules/no-unknown-property.js [hunk 2]
```diff
@@ -230,7 +237,12 @@ function getDOMPropertyNames(context) {
...
+  }
+  // these were added in React v16.4.0, see https://reactjs.org/blog/2018/05/23/react-v-16-4.html and https://github.com/facebook/react/pull/12507
+  if (testReactVersion(context, '>= 16.4.0')) {
+    return ALL_DOM_PROPERTY_NAMES.concat(['onPointerDown', 'onPointerMove', 'onPointerUp', 'onPointerCancel', 'onGotPointerCapture', 'onLostPointerCapture',
```
Comment (ljharb) [inline lib/rules/no-unknown-property.js:244]:
```suggestion
    return ALL_DOM_PROPERTY_NAMES.concat('onPointerDown', 'onPointerMove', 'onPointerUp', 'onPointerCancel', 'onGotPointerCapture', 'onLostPointerCapture',
```

File: lib/rules/no-unknown-property.js [hunk 3]
```diff
@@ -230,7 +237,12 @@ function getDOMPropertyNames(context) {
...
+      'onPointerEnter', 'onPointerLeave', 'onPointerOver', 'onPointerOut']);
```
Comment (ljharb) [inline lib/rules/no-unknown-property.js:245]:
```suggestion
      'onPointerEnter', 'onPointerLeave', 'onPointerOver', 'onPointerOut');
```

**Review by ljharb — COMMENTED** [review]
Thanks!

**Comment by sjarva** [comment]
I made the suggestions you had ✔️

**Comment by jdufresne** [comment]
My project starting emitting eslint errors when using touch events. I don't see them events listed in this change. Could/should they also be included? 

https://reactjs.org/docs/events.html#touch-events

- onTouchCancel
- onTouchEnd
- onTouchMove
- onTouchStart

**Comment by sjarva** [comment]
Thanks for pointing them out @jdufresne 👍 
I added touch event and media related properties to this PR.

**Review by ljharb — APPROVED** [review]

**Pull request merged**